### PR TITLE
Courses unit tests

### PR DESCRIPTION
**Pull request by msladwig**
Added unit tests for course and additional fields to models to reflect use cases.

### CODE REVIEW

**Review by Svedek — CHANGES_REQUESTED** [review]
the assertRaises used don't work the way they are implemented, I will write notes on them for proper implementation and continue looking over the test case.

File: schedulingApp/unit-tests/test_courses.py [hunk 1]
```diff
@@ -1,38 +1,126 @@
...
+                         msg="Should not allow multiple courses with the same semester and title")
+
+    def test_sameNameDifferentSemester(self):
+        self.setUp()
+        a = Course("CS423", "Spring 2023")
+        a.title = "CS423"
+        a.semester = "Spring 2023"
+        a.setTitle("CS361")
+        self.assertEqual(a.title, "CS361",
+                         msg="Should update to the same name as a different course as long as semesters are different.")
```
Comment (Svedek) [inline schedulingApp/unit-tests/test_courses.py:106]: Should also test sameNameSameSemester has no change/raises error

File: schedulingApp/unit-tests/test_courses.py [hunk 2]
```diff
@@ -1,38 +1,126 @@
...
+        self.assertRaises(NameError, self.myCourse.__del__(),
+                          msg="Trying to delete nonexistent course should throw NameError")
 
     def test_validAndExistingCourse(self):
-        pass
+        self.setUp()
+        self.theCourse.__del__()
+        self.assertFalse(self.theCourse.getTitle() == "CS361", msg="Title for deleted course should no longer exist.")
+        self.assertFalse(self.theCourse.getSemester() == "Fall 2022",
+                         msg="Semester for deleted course should no longer exist.")
```
Comment (Svedek) [inline schedulingApp/unit-tests/test_courses.py:126]: Not sure about this one, but if delete is called, shouldnt getTitle and getSemester error, ill bring this up at the meeting tomorrow to be sure

File: schedulingApp/unit-tests/test_courses.py [hunk 3]
```diff
@@ -1,38 +1,126 @@
...
-
+from models import Course
 
 # Fields: Title
 class CreateCourse(TestCase):
-    def test_emptyTitle(self):
-        pass
+    def test_noArgs(self):
+        a = Course()
+        self.assertRaises(TypeError, msg="Should raise TypeError for no arguments, requires a title and semester")
```
Comment (Svedek) [inline schedulingApp/unit-tests/test_courses.py:8]: The assertRaises doesn't work after the error is raised, proper implementation would look as follows:

```suggestion
        with self.assertRaises(TypeError, msg="Should raise TypeError for no arguments, requires a title and semester"):
                a = Course()
```

**Review by Svedek — CHANGES_REQUESTED** [review]

File: schedulingApp/models.py [hunk 4]
```diff
@@ -20,6 +20,26 @@ class User(models.Model):
 
 class Course(models.Model):
     title = models.CharField(max_length=32)
+    semester = models.CharField(max_length=32)
```
Comment (MWHunter) [inline schedulingApp/models.py:23]: lgtm

File: schedulingApp/unit-tests/test_courses.py (hunk 1 above)
Comment (Svedek) [reply schedulingApp/unit-tests/test_courses.py:106]: Edit: this has been checked, ignore

File: schedulingApp/unit-tests/test_courses.py (hunk 2 above)
Comment (msladwig) [reply schedulingApp/unit-tests/test_courses.py]: Updated deletion testing, should look better now.

**Pull request closed without merging**